### ISSUE TITLE
[Bug] scale_inv may be not synchronized in TP group

### ISSUE THREAD

**Issue by wkcn** [NVIDIA/TransformerEngine#335]
Hi there, I found a bug when training a GPT model with the latest Megatron-LM and Transformer Engine v0.3~v0.10.

When Tensor Parallellism (TP) > 1, the training loss curve (pink color) is abnormal. However, the loss curve (orange color) is normal when TP=1.

FP8 and Sequence Parallelism are both enabled in the two experiments.

![image](https://github.com/NVIDIA/TransformerEngine/assets/3415417/6137243d-e5e3-4f91-bf74-0f30b2727417)

After reading the code of TE, I guess that `scale_inv` may be not synchronized in TP group.
For example, when using Sequence Parallelism, the FP8 input is gathered in TP group. I'm not sure whether the TP group shares the same `scale_inv`. (https://github.com/NVIDIA/TransformerEngine/blob/main/transformer_engine/pytorch/module/linear.py#L127).

**Comment by Marks101** [comment]
We encountered similar issues with Megatron-LM. This did the job for us: https://github.com/NVIDIA/Megatron-LM/pull/442

**Comment by ksivaman** [comment]
This issue is resolved in megatron-LM [TOT](https://github.com/NVIDIA/Megatron-LM/tree/9beed7a0d50d5f18c27c63d6941d403ae4d3d1cb). Thanks @Marks101!

**Comment by wkcn** [comment]
Thanks @Marks101 and @ksivaman !
